catalog/dags/data_refresh/reporting.py

```diff
@@ -30,9 +30,29 @@ def report_record_difference(before: str, after: str, media_type: str, dag_id: s
 Data refresh for {media_type} complete! :tada:
 _Note: All values are row estimates and are not (but nearly) exact_
 *Record count difference for `{media_type}`*: {before:,} → {after:,}
-*Change*: {count_diff:+,} ({percent_diff:+}% Δ)
+*Change*: {count_diff:+,} ({_round_sig_figs(percent_diff, 3):+}% Δ)
 """
     slack.send_message(
         text=message, dag_id=dag_id, username="Data refresh record difference"
     )
     return message
+
+
+def _round_sig_figs(num, n):
+    """
+    Round number to the nearest n digits.
+    If num is an integer, don't change it.
+    If num has a decimal, round to 3 significant figures (after any zeros),
+    rounding up if the next digit is 5 or greater.
+    """
+
+    if num % 1 == 0:
+        return int(num)
+
+    # Round decimal part to 3 sig figs. Leave int part untouched.
+    [int_part, dec_part] = str(num).split(".")
+
+    # Have to awkwardly prepend the decimal part with '0.' to get correct
+    # rounding behavior, then slice '0.' off before concatenating
+    # the final number.
+    return float(int_part + "." + "{:.{n}g}".format(float("0." + dec_part), n=n)[2:])
```

Comment:
This is over complicating things a bit, I think for the sake of always having 3 significant digits. Using some fancy [string format spec](https://docs.python.org/3/library/string.html#format-specification-mini-language), the whole thing can be simplified, so long as we don't care that `0.00005646%` will get truncated to just `0.00006%`... and frankly, at such small percentages, I really don't think it matters to have accuracy past 1/100, it just isn't interesting or useful information to have such abstract and small percentages.

I've left a suggestion on the line above for format spec to use.

Reply:
@sarayourfriend I was talking about this with @mattfergoda in person - many times for the data refresh, we do have changes that are quite small by percentage but still register as a +/- half a million record difference. Things like `0.0654%` or even `0.00481%`. For me, and perhaps this is just my own perspective, truncating those down completely does lose some of the specificity we have. Maybe it doesn't matter, but whatever change we make I'd like to _at least_ be sure we have a few digits available a couple of decimal points out.

Reply:
What do you mean by "truncating them down completely"? As I wrote:

> 0.00005646% will get truncated to just 0.00006%

In what way is this a meaningful difference in the information, when the actual count is also present?

Reply:
Those are all fair points! We can go with the much simpler implementation with that in mind 🙂